sdk/tables/data-tables/src/TablesBatch.ts

```diff
@@ -0,0 +1,321 @@
+      return rawBatchResponse;
+    }
+  };
+}
+
+/**
+ * This method creates a batch request object that provides functions to build the envelope and body for a batch request
+ * @param batchGuid Id of the batch
+ */
+function createInnerBatchRequest(batchGuid: string) {
+  const HTTP_LINE_ENDING = "\r\n";
+  const HTTP_VERSION_1_1 = "HTTP/1.1";
+  const changesetGuid = generateUuid();
+
+  // batch_{batchid}
+  const batchBoundary = `batch_${batchGuid}`;
+  const changesetBoundary = `changeset_${changesetGuid}`;
+
+  const subRequestPrefix = `--${changesetBoundary}${HTTP_LINE_ENDING}${HeaderConstants.CONTENT_TYPE}: application/http${HTTP_LINE_ENDING}${HeaderConstants.CONTENT_TRANSFER_ENCODING}: binary`;
+  // Content-Type: multipart/mixed; boundary=changeset_{changesetGuid}
+  const multipartContentType = `multipart/mixed; boundary=changeset_${changesetGuid}`;
+  const chaangesetEnding = `--${changesetBoundary}--`;
+  const batchEnding = `--${batchBoundary}`;
+
+  return {
+    body: `--${batchBoundary}${HTTP_LINE_ENDING}${HeaderConstants.CONTENT_TYPE}: multipart/mixed; boundary=changeset_${changesetGuid}${HTTP_LINE_ENDING}${HTTP_LINE_ENDING}`,
+    operationCount: 0,
+    createPipeline() {
+      const policyFactoryLength = 3;
+      const factories: RequestPolicyFactory[] = new Array(policyFactoryLength);
+
+      factories[0] = deserializationPolicy();
+      factories[1] = new BatchHeaderFilterPolicyFactory();
+
+      factories[policyFactoryLength - 1] = new BatchRequestAssemblePolicyFactory(this); // Use batch assemble policy to assemble request and intercept request from going to wire
+      return factories;
+    },
+    appendSubRequestToBody(request: WebResource) {
+      // Start to assemble sub request
+      this.body += [
+        subRequestPrefix, // sub request constant prefix
+        "", // empty line after sub request's content ID
+        `${request.method.toString()} ${getURLPathAndQuery(
+          request.url
+        )} ${HTTP_VERSION_1_1}${HTTP_LINE_ENDING}` // sub request start line with method,
+      ].join(HTTP_LINE_ENDING);
+
+      for (const header of request.headers.headersArray()) {
+        this.body += `${header.name}: ${header.value}${HTTP_LINE_ENDING}`;
+      }
+
+      this.body += `${HTTP_LINE_ENDING}${request.body}${HTTP_LINE_ENDING}`; // sub request's headers need end with an empty line
+      this.operationCount++;
+    },
+    getHttpRequestBody(): string {
+      return `${this.body}${HTTP_LINE_ENDING}${chaangesetEnding}${HTTP_LINE_ENDING}${batchEnding}${HTTP_LINE_ENDING}`;
+    },
+    getMultipartContentType(): string {
+      return multipartContentType;
+    },
+    getBatchBoundary(): string {
+      return batchBoundary;
+    }
+  };
+}
+
+/**
+ * Get URL path and query from an URL string.
+ *
+ * @export
+ * @param {string} url Source URL string
+ * @returns {(string | undefined)}
+ */
+export function getURLPathAndQuery(url: string): string | undefined {
```

Comment:
I don't understand what this function does. It seems like it strips off the queryString, but the name doesn't make sense to me. Also what's the point in sanitizing the queryString if it's not returned or used?